### PR TITLE
[Grid] Field, Action & Filter position in Grids

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | related to #7008 |
| License         | MIT |

By changing the value of `position` on field/filter/action we can organize them in a grid. They are now rendered sorted by position ASC. **Position defaults to `100`** to ensure that default will be rendered after the configured.

### CODE REVIEW

File: src/Sylius/Component/Grid/spec/Definition/FieldSpec.php [hunk 1]
```diff
@@ -91,4 +91,12 @@ function it_can_have_options()
         $this->setOptions(['template' => 'SyliusUiBundle:Grid/Field:_status.html.twig']);
         $this->getOptions()->shouldReturn(['template' => 'SyliusUiBundle:Grid/Field:_status.html.twig']);
     }
+
+    function it_has_position_no_position_by_default()
```
Comment (tuka217) [inline src/Sylius/Component/Grid/spec/Definition/FieldSpec.php]: `it_has_no_position_by_default`. And there should be two methods: `no_position_by_default` and `mutable_position`

File: src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% macro headers(grid, definition, requestAttributes) %}
     {% import 'SyliusUiBundle:Macro:sorting.html.twig' as sorting %}
 
-    {% for field in definition.fields %}
+    {% for field in definition.fields|sort_by('position', 'DESC') %}
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig]: I think it should be ASC, hm?

Comment (TheMadeleine) [inline src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig]: Then what value should be default on the Field model?

File: src/Sylius/Component/Grid/spec/Definition/FieldSpec.php (hunk 1 above)
Comment (TheMadeleine) [inline src/Sylius/Component/Grid/spec/Definition/FieldSpec.php]: Yeah, moths of docs writing and you forget how to spec. 😞

Comment (tuka217) [inline src/Sylius/Component/Grid/spec/Definition/FieldSpec.php]: :(

File: src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig (hunk 2 above)
Comment (pjedrzejewski) [inline src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig]: 0 is OK.

File: src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {% macro headers(grid, definition, requestAttributes) %}
     {% import 'SyliusUiBundle:Macro:sorting.html.twig' as sorting %}
 
-    {% for field in definition.fields %}
+    {% for field in definition.fields|sort_by('position', 'ASC') %}
```
Comment (GSadee) [inline src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig]: By default is `ASC`, you can remove second parameters

Comment (TheMadeleine) [inline src/Sylius/Bundle/UiBundle/Resources/views/Macro/table.html.twig]: That's true!

**Comment by pjedrzejewski** [comment]
Thanks Magda!

**Pull request merged**